osu.Game/Overlays/Comments/GetCommentRepliesButton.cs

```diff
@@ -0,0 +1,94 @@
+﻿// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Graphics;
+using osu.Game.Graphics;
+using osu.Framework.Graphics.Sprites;
+using osu.Game.Online.API.Requests.Responses;
+using osu.Framework.Allocation;
+using osu.Game.Graphics.UserInterface;
+using System.Collections.Generic;
+using osuTK;
+using osu.Game.Online.API;
+using osu.Framework.Bindables;
+using osu.Game.Online.API.Requests;
+using System.Linq;
+
+namespace osu.Game.Overlays.Comments
+{
+    public abstract class GetCommentRepliesButton : LoadingButton
+    {
+        public readonly Bindable<List<Comment>> ChildComments = new Bindable<List<Comment>>();
+        public readonly Bindable<CommentsSortCriteria> Sort = new Bindable<CommentsSortCriteria>();
+
+        protected override IEnumerable<Drawable> EffectTargets => new[] { text };
+
+        protected readonly Comment Comment;
+
+        [Resolved]
+        private IAPIProvider api { get; set; }
+
+        private SpriteText text;
+        private GetCommentRepliesRequest request;
+        private int currentPage;
+
+        protected GetCommentRepliesButton(Comment comment)
+        {
+            Comment = comment;
+
+            AutoSizeAxes = Axes.Both;
+            LoadingAnimationSize = new Vector2(8);
+            Action = onAction;
+        }
+
+        protected override void LoadComplete()
+        {
+            base.LoadComplete();
+            ChildComments.BindValueChanged(OnChildrenChanged, true);
+        }
+
+        protected virtual void OnChildrenChanged(ValueChangedEvent<List<Comment>> children)
+        {
+        }
+
+        protected override Container CreateBackground() => new Container
+        {
+            AutoSizeAxes = Axes.Both
+        };
+
+        protected override Drawable CreateContent() => text = new SpriteText
+        {
+            AlwaysPresent = true,
+            Font = OsuFont.GetFont(size: 12, weight: FontWeight.Bold),
+            Text = ButtonText(),
+        };
+
+        protected abstract string ButtonText();
+
+        private void onAction()
+        {
+            request = new GetCommentRepliesRequest(Comment.Id, Sort.Value, ++currentPage);
+            request.Success += onSuccess;
+            api.Queue(request);
+        }
+
+        private void onSuccess(CommentBundle response)
+        {
+            var children = ChildComments.Value.ToList();
+            response.Comments.ForEach(c =>
+            {
+                if (c.ParentId == Comment.Id && children.Count(u => u.Id == c.Id) == 0)
```

Comment:
* Why is the parent id checked here? I assume web should take care of that, since it's passed in the request.
* `Count` is slightly inefficient here, since it won't stop at the first matching result. Use negated `.Any()` or `.All()` instead.

Reply:
I could just use `Contains` but this is not working because some data can be differ for one comment but different requests (for example, in one request comment don't have children, but in another it has)

Reply:
We can receive the same comment twice (first time in the main request, second - in the `GetRepliesRequest`)